### ISSUE TITLE
Angular2-meteor doesn't work inside a ionic2-rc project.

### ISSUE THREAD

**Issue by mmazloum** [Urigo/angular-meteor#1460]
Hi,

I'm trying to use angular2-meteor inside an ionic2-rc project, the package bundler they prefer is rollup but I get lots of error just when I try to 

`import {METEOR_PROVIDERS} from 'angular2-meteor';`

My try was to do like the whatsapp2 toturial but I ran into lots of errors like those : 

```
 [22:25:31]  transpile update started ... 
[22:25:33]  typescript: node_modules/angular2-meteor/dist/index.d.ts, line: 1 
            File '/Users/marwanmazloum/Desktop/Developpement/TF1/tf1-app/node_modules/angular2-meteor/typings/index.d.ts' 
            not found. 

       L1:  export * from './zone_utils';
       L2:  export * from './mongo_cursor_observer';

[22:25:33]  typescript: node_modules/angular2-meteor/dist/meteor_reactive.d.ts, line: 23 
            Cannot find namespace 'Tracker'. 

      L23:  autorun(func: (c: Tracker.Computation) => any, autoBind?: Boolean): Tracker.Computat

[22:25:33]  typescript: node_modules/angular2-meteor/dist/meteor_reactive.d.ts, line: 23 
            Cannot find namespace 'Tracker'. 

      L23:  autorun(func: (c: Tracker.Computation) => any, autoBind?: Boolean): Tracker.Computation;

[22:25:33]  typescript: node_modules/angular2-meteor/dist/meteor_reactive.d.ts, line: 29 
            Cannot find namespace 'Meteor'. 

[22:25:33]  typescript: node_modules/angular2-meteor/dist/mongo_cursor_differ.d.ts, line: 33 
[22:25:33]  typescript: node_modules/angular2-meteor/dist/mongo_cursor_observer.d.ts, line: 35 
      L29:  subscribe(name: string, ...args: any[]): Meteor.SubscriptionHandle;
      L30:  call(name: string, ...args: any[]): any;

            Cannot find namespace 'Mongo'. 

      L32:  forEachOperation(fn: Function): void;
      L33:  diff(cursor: Mongo.Cursor<any>): this;
      L34:  onDestroy(): void;

            Cannot find namespace 'Mongo'. 

      L34:  static isCursor(cursor: any): boolean;
      L35:  constructor(cursor: Mongo.Cursor<any>, _debounceMs?: number);
      L36:  subscribe(events: any): any;

[22:25:33]  typescript: node_modules/angular2-meteor/dist/zone_utils.d.ts, line: 16 
            Cannot find name 'Zone'. 

[22:25:33]  typescript: node_modules/angular2-meteor/dist/zone_utils.d.ts, line: 18 
[22:25:33]  typescript: node_modules/angular2-meteor/dist/zone_utils.d.ts, line: 19 
[22:25:33]  typescript: node_modules/angular2-meteor/dist/zone_utils.d.ts, line: 20 
[22:25:33]  typescript: node_modules/angular2-meteor/dist/zone_utils.d.ts,
```

**Comment by Urigo** [comment]
yes we are aware of this issue, this is why only the 2.0 WhatsApp tutorial with the Meteor CLI is working and the Ionic CLI is not updated yet. (https://www.angular-meteor.com/tutorials/whatsapp2/)
We are not the only ones with this problem so Ionic is working on a new version to fix this and then we will update the tutorial and make it work

**Comment by mmazloum** [comment]
Cool I'll try form the socially-ionic example then.

however the whatsapp2 and angular2 socialy projects enable me to do lots of cool things ;-)
thanks a lot for your work.

**Comment by sosnet** [comment]
Is this related to **Webpack**? As they published a beta that's include it again: https://forum.ionicframework.com/t/help-us-test-webpack-support-in-ionic-2-app-scripts/67507

**Comment by mmazloum** [comment]
Thanks for the hint i'll try and give feedback here, did you try already ? I'll also need to try by adding ionic2 to a meteor angular2 project since I find the Meteor-Angular2 structure very interesting to.

The thing Is that I use a separate meteor server that I connect through DDP to my app, so my Mongo collections are declared differently and I use the DDP object.

**Comment by sosnet** [comment]
Maybe @Urigo can have a look on this beta, perhaps an update of meteor and the tutorial is already possible. :+1:

**Comment by ekhmoi** [comment]
+1 waiting for update

**Comment by Urigo** [comment]
it looks like it can work now but I must say, all those breaking changes are too much for us right now (and also not fair for the community).
We will wait for a few more RC releases and then update the tutorial.
If someone wants to try in the meantime, we will help and support you (you can contact me directly on my Github profile)

**Comment by dotansimha** [comment]
You can use Ionic 2 rc1, with Webpack and custom build, more info here: https://github.com/idanwe/meteor-client-side/issues/31#issuecomment-258114398

Closing it since it's now possible to use both together, soon the tutorial will be deployed (at the moment, it's updated code is available in https://github.com/DAB0mB/ionic2-meteor-messenger/tree/step-by-step-new)

**Comment by sosnet** [comment]
Can we get notified here when the tutorial is updated?
Thanks for the code @dotansimha, I'll have a look into it tomorrow!

**Comment by dotansimha** [comment]
The tutorial is ready and in PR: https://github.com/Urigo/angular-meteor-docs/pull/284
@Urigo will soon merge and deploy to angular-meteor.com :)

**Comment by Urigo** [comment]
merged and deployed!
amazing work @dotansimha !

**Comment by sosnet** [comment]
Hey @dotansimha. Thank you again.
I started with the tutorial and I'm at the end of Step 2 (https://www.angular-meteor.com/tutorials/whatsapp2/ionic/chats-page)
Unfortunately I'm getting an error in the console saying **Cannot find module "../pages/chats/chats"**
When I try **ionic run** I'm getting the error  _Cannot find name 'Chat'_. Seems like there is an error with including those models with the alias **api**.

Are those steps in **Step 1 - Initial setup** are really necessary? I thought that the newest version of ionic2 is already using _webpack_ in addition? I'm asking because I also tried to adapt my existing project to the tutorial, but it's not compiling my changes anymore but also not showing an error at the moment .. Only in ionic run, as with the tutorial.

**Comment by dotansimha** [comment]
@sosnet, can you share you repo so I can take a look at the error? 

Also, the Step 1 is really critical because Ionic 2 uses Rollup (with es2016 modules), and in this step we replace Rollup with Webpack and CommonJS modules.

**Comment by sosnet** [comment]
Ok, looks like there was another update of the tutorial for Step 1. Anyway, now Step 1 and 2 are working for me, so great!
But when I start the app after Step 3, the app shows only white screen and in the browser-terminal I get this error:
`webpack:///./api/collections/whatsapp-collections.ts?:1 Uncaught Error: Module build failed: TypeError: Cannot read property 'content' of undefined(…)`

@dotansimha, the repo is https://github.com/sosnet/ionic2-meteor
Thank you so much for helping!

**Comment by scorpio13love** [comment]
I'm trapped in the same situation as sosnet. And what the correct  "webpack.config.js" like?

**Comment by kamilkisiela** [comment]
@sosnet It's not angular-meteor specific. [Here](https://github.com/sosnet/ionic2-meteor/blob/master/config/webpack.config.js#L59-L62) is a line that causes an error. The thing is, it's not working with aliases.

You can just disable it to make it works but you will skip the features of `typescript-sourcemap-loader-memory`.

**Comment by sosnet** [comment]
Hey @kamilkisiela, thank you very much for helping. I'm new to Angular2/Ionic2 and Meteor.. So, would you be so kind and tell me how to disable it? Or what are those features I would lose? Are they important?

Or should I remove those aliases? Or are they needed later?
Sorry for all those noob questions..

**Comment by kamilkisiela** [comment]
@sosnet It's basically for the source maps. Here's [the actual file](https://github.com/driftyco/ionic-app-scripts/blob/master/src/webpack/typescript-sourcemap-loader-memory.ts).

I think you can skip it and use just `devtool: 'source-map'`.

Here's how the `webpack.config.js` would look like:

```js
var path = require('path');
var webpack = require('webpack');

var ionicWebpackFactoryPath = path.join(process.env.IONIC_APP_SCRIPTS_DIR, 'dist', 'webpack', 'ionic-webpack-factory.js');
var ionicWebpackFactory = require(ionicWebpackFactoryPath);

function getEntryPoint() {
  if (process.env.IONIC_ENV === 'prod') {
    return '{{TMP}}/app/main.prod.js';
  }
  return '{{SRC}}/app/main.dev.ts';
}

/*
function getPlugins() {
  if (process.env.IONIC_ENV === 'prod') {
    return [
      // This helps ensure the builds are consistent if source hasn't changed:
      new webpack.optimize.OccurrenceOrderPlugin(),

      // Try to dedupe duplicated modules, if any:
      // Add this back in when Angular fixes the issue: https://github.com/angular/angular-cli/issues/1587
      //new DedupePlugin()
    ];
  }

  // for dev builds, use our custom environment
  return [
    ionicWebpackFactory.getIonicEnvironmentPlugin()
  ];
}
*/

function getPlugins() {
  var plugins = [
    // Try to dedupe duplicated modules, if any:
    // Add this back in when Angular fixes the issue: https://github.com/angular/angular-cli/issues/1587
    //new DedupePlugin()
    new webpack.ProvidePlugin({
      __extends: 'typescript-extends'
    })
  ];

  if (process.env.IONIC_ENV === 'prod') {
    // This helps ensure the builds are consistent if source hasn't changed:
    plugins.push(new webpack.optimize.OccurrenceOrderPlugin())
  }

  return plugins;
}

function getDevtool() {
  return process.env.IONIC_SOURCE_MAP;
}

module.exports = {
  entry: getEntryPoint(),
  output: {
    path: '{{BUILD}}',
    filename: 'main.js'
  },
  devtool: getDevtool(),

  resolve: {
    extensions: ['.js', '.json', '.ts'],
    alias: {
      'api': path.resolve(__dirname, '../api')
    }
  },

  module: {
    loaders: [
      {
        test: /\.json$/,
        loader: 'json'
      },
      {
        test: /\.ts$/,
        exclude: /(node_modules)/,
        loaders: ['awesome-typescript-loader']
      }
    ]
  },

  plugins: getPlugins(),

  // Some libraries import Node modules but don't use them in the browser.
  // Tell Webpack to provide empty mocks for them so importing them works.
  node: {
    fs: 'empty',
    net: 'empty',
    tls: 'empty',
  __dirname: true
  }
};
```

**Comment by sosnet** [comment]
Thx for your effort @kamilkisiela, but unluckily I'm getting this error
`http://localhost:8100/tabs.html Failed to load resource: the server responded with a status of 404 (Not Found)`

**Comment by kamilkisiela** [comment]
Please take a look at https://github.com/Urigo/Ionic2CLI-Meteor-WhatsApp and compare `package.json`, `config/webpack.config.js`. Maybe there are some differences? It's Ionic who is in charge of parsing your templates into strings.

**Comment by sosnet** [comment]
Ok, back to the roots.
I started now with `git clone https://github.com/Urigo/Ionic2CLI-Meteor-WhatsApp`. The biggest differences  are the `config/webpack.config.js` and some downgrades of packages in `package.json`. I think the webpack.config should be included to the tutorial, otherwise it's useless.

But, the app is starting now, but I get those errors in the browser console:
```
Failed to load resource: the server responded with a status of 404 (Not Found): http://localhost:8100/sockjs/info?cb=sg98u3njre
```
When I open the url and change the port to `3000` I get something back. But how to tell the ionic app which port and url it should use for the meteor-server?

Thanks a lot, @kamilkisiela, I'm a step closer again.

**Comment by sosnet** [comment]
Ok, I solved this problem by adding these lines to the `index.html` of my ionic project:
```
<script>
      __meteor_runtime_config__ = {
        DDP_DEFAULT_CONNECTION_URL: 'http://localhost:3000'
      };
</script>
```

Why it's not using it's default settings? Why I have to set them and why the settings in `node_modules/meteor-client-side/meteor-runtime-config.js` are ignored?

However, I will continue now with the tutorial where I stopped before. Thanks everyone who helped me out. Maybe someone can use those problems and solutions to fix something.
